react/Gallery.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
```

Comment:
Careful with the renaming of `Gallery.js` to `GalleryLegacy.js`. People could be using this component through:
`import { Gallery } from 'vtex.search-result'`
In that case, they would now be importing **this** code, and not the legacy one

Reply:
From this code, it looks fine to me, and I see that you're taking that into account! Pointing this out just to keep in mind :)
I recommend linking this branch in `carrefour` and `motorola`, since they're the most likely to be customizing something with this component.